From: David Hildenbrand <david@redhat.com>
Subject: fixup: mm: let _folio_nr_pages overlay memcg_data in first tail page
Date: Wed, 5 Mar 2025 11:14:52 +0100

Make "make htmldoc" happy by marking non-private placeholder entries
private.

Link: https://lkml.kernel.org/r/a97f8a91-ec41-4796-81e3-7c9e0e491ba4@redhat.com
Signed-off-by: David Hildenbrand <david@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/mm_types.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/linux/mm_types.h~mm-let-_folio_nr_pages-overlay-memcg_data-in-first-tail-page-fix
+++ a/include/linux/mm_types.h
@@ -381,18 +381,20 @@ struct folio {
 		struct {
 			unsigned long _flags_1;
 			unsigned long _head_1;
-	/* public: */
 			union {
 				struct {
+	/* public: */
 					atomic_t _large_mapcount;
 					atomic_t _entire_mapcount;
 					atomic_t _nr_pages_mapped;
 					atomic_t _pincount;
+	/* private: the union with struct page is transitional */
 				};
 				unsigned long _usable_1[4];
 			};
 			atomic_t _mapcount_1;
 			atomic_t _refcount_1;
+	/* public: */
 #ifdef NR_PAGES_IN_LARGE_FOLIO
 			unsigned int _nr_pages;
 #endif /* NR_PAGES_IN_LARGE_FOLIO */
_
